### ISSUE TITLE
Big picture mode always opens on wrong screen

### ISSUE THREAD

**Issue by ChrisLauinger77** [ValveSoftware/steam-for-linux#4178]
I have 2 screen setup. both same 1080p displays.
I made the left one the primary one with nvidia-settings.
Gnome display tool shows the left one as index 2 and the right one as index 1.
https://dl.dropboxusercontent.com/u/44733397/gnome-display-settings.png

The games usually start on the primary left one (except dota 2 reborn which i needed to add a launch option "-sdl_displayindex 2".
When I press the big picture button in steam client it always opens the big picture on the right one (my steam link however displays the primary left one), which is annoying because i need to walk to the pc and switch the big picture mode steam from right to left display.

---

Processor Information:
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x2a
    CPU Stepping:  0x7
    CPU Type:  0x0
    Speed:  5900 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    Debian GNU/Linux testing-updates (sid) (64 bit)
    Kernel Name:  Linux
    Kernel Version:  4.2.0-1-amd64
    X Server Vendor:  The X.Org Foundation
    X Server Release:  11703000
    X Window Manager:  GNOME Shell
    Steam Runtime Version:  steam-runtime-release_2015-06-12

Video Card:
    Driver:  NVIDIA Corporation GeForce GTX 770/PCIe/SSE2

```
Driver Version:  4.5.0 NVIDIA 352.63
OpenGL Version: 4.5
Desktop Color Depth: 24 bits per pixel
Monitor Refresh Rate: 60 Hz
VendorID:  0x10de
DeviceID:  0x1184
Number of Monitors:  2
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 3840 x 1080
Primary Display Size: 20,91" x 11,77"  (23,98" diag)
                                        53,1cm x 29,9cm  (60,9cm diag)
Primary Bus: PCI Express 16x
Primary VRAM: 2048 MB
Supported MSAA Modes:  2x 4x 8x 16x 
```

Sound card:
    Audio device: AV200

Memory:
    RAM:  7953 Mb

Miscellaneous:
    UI Language:  English
    LANG:  de_DE.UTF-8
    Microphone:  Not set
    Total Hard Disk Space Available:  53884 Mb
    Largest Free Hard Disk Block:  22492 Mb

Installed software:

Recent Failure Reports:

**Comment by Kallestofeles** [comment]
Any progress on this?
This currently happens on Mate and Gnome desktop environments. On XFCE, the issue disappears.

**Comment by asim-vax** [comment]
Happens on ubuntu gnome 17.04, monitor 1 is set to primary, big picture opens on monitor 2, game opens on monitor 1 when launched

**Comment by KevinOberle** [comment]
Me too. Exactly this. On latest SolusOS.

I'm using an AMD RX 470 video card, i5-7500 cpu.

**Comment by KevinOberle** [comment]
I should add: this bug breaks steam link functionality. The steam link zooms in on the primary monitor meanwhile big picture is open on the secondary. You have to go to the pc and manually launch the game you want.

**Comment by asim-vax** [comment]
Will there be an update on this? With ubuntu moving from unity to gnome this breaks steam hardware viability.

**Comment by nlfog** [comment]
Same issue here on Ubuntu 18.04

2 Monitors, left one is marked primary in nvidia settings and gnome settings.  Launching big picture mode launches on the right screen.  This completely breaks my steamlink as only my empty left desktop screen is mirrored to my tv.

**Comment by asim-vax** [comment]
@nlfog oddly, mine started working on a fresh ubuntu 18.04 install (launches to the correct monitor) did you upgrade or fresh install?

**Comment by nlfog** [comment]
@asim-vax Fresh install of 18.04 .  I've also just tried to steam beta client as well, no change.  Not sure what to check, as everything I can see so far is setting the left as primary monitor.  And, like others have reported, big screen opens on the right while games launch on the left.

**Comment by asim-vax** [comment]
Yeah, it happened to me from 17.04/17.10/18.04, but i did a fresh install and suddenly it started working, where did you install steam from, the provided software GUI or from the steam website? This was the only thing i changed from memory

**Comment by asim-vax** [comment]
@nlfog i installed from the software centre, because at the time the deb from the website had issues - try the software centre

**Comment by Solarinass** [comment]
I'm on debian testing and thig bus persists for me. I'm using the packages in the debian repos

It's kind of odd that it will always use the second monitor instead of the primary one and kind of annoying that I have to move it all the time. I would be nice to see a feature implemented that allows linux users to pick which monitor it starts on and have it default to the primary one

**Comment by Snepsts** [comment]
I, too, have been experiencing this on Ubuntu 18.04 with nvidia-396.54 drivers.

I tested and found that if you make your main monitor the "tallest monitor", it will start on that monitor. (I put my left and right monitor slightly lower in the GUI placement of the screen, so they start at 40 or something instead of 0 on the y axis)

Weird fix but it works for me so far.

**Comment by ForsakenNGS** [comment]
> I tested and found that if you make your main monitor the "tallest monitor", it will start on that monitor. (I put my left and right monitor slightly lower in the GUI placement of the screen, so they start at 40 or something instead of 0 on the y axis)
> 
> Weird fix but it works for me so far.

I can confirm this with the current Arch Linux version and gnome 3.30.0-1. (uname 4.18.16-arch1-1-ARCH #1 SMP PREEMPT)

I am using two full hd monitors. When configuring the second monitor with "+1920+26" big picture still opens at that, at "+1920+27" or lower it will properly use the primary monitor. Comparing this with the height of the top menubar (or whatever it is called) of gnome makes me guess steam detects the primary monitor starting lower than the secondary because of that and decides to launch on the monitor that is farther on the top. Resetting the offset and switching the primary monitor setting imho confirmed it.
Basically if you have two full hd monitors horizontally without any vertical offset, steam will always launch on the one WITHOUT the menu bar.

I will go for the "+1920+27" workaround and hope this gets fixed soon. (Remember, if I'm right the required vertical offset may vary depending on your desktop manager/configuration)

**Comment by Snepsts** [comment]
This makes a lot of sense, actually. Thanks for the explanation, Forsaken!

One interesting thing to point out is this is not an issue on xfce with a top menu bar. Haven't tested others.

**Comment by 8BitCerberus** [comment]
> I will go for the "+1920+27" workaround and hope this gets fixed soon. (Remember, if I'm right the required vertical offset may vary depending on your desktop manager/configuration)

Confirming this for Ubuntu Budgie 18.04.1. I've just been setting both monitors to 1080 and BPM has been launching on the primary/1st monitor for months now. But for some reason yesterday it started back with BPM launching on the 2nd monitor (which is index 1, even though the 1st monitor is marked as primary, though index 2). I set my offset +100 for the 2nd monitor and BPM launches on the 1st every time as it's supposed to, again. I will test to see if +27 will work for me as well, I was just trying to get a game going for game night on my Steam Link and went for a higher number to be sure, instead of spending more time troubleshooting.

**Comment by DyXel** [comment]
Any update on this? We should get an option to choose the specific display we want big picture to start on, regardless of configuration. Either that or a fix for this weird behaviour.

**Comment by qfettes** [comment]
This is a persistent issue for me using steam link + BPM for my Ubuntu 18.04 machine. The ability to select the monitor BPM launches on would solve this and add a useful feature.

**Comment by orkanaye** [comment]
same here. makes steamlink unusable. Also full screen videos from steam launch on wrong screen as well as dota2.

Most of other games start properly on the primary monitor.

Ubuntu 18.04 /w cinnamon. monitors are different models

**Comment by jackmawer** [comment]
This issue *still* exists! Why doesn’t big picture mode have an option to choose the starting monitor?

**Comment by treier57lol** [comment]
Still an issue on Ubuntu 18.04 with KDE Plasma. 
> Why doesn’t big picture mode have an option to choose the starting monitor?

It does, on Windows. There shouldn't be an excuse for the option not being in Linux.

**Comment by kforney** [comment]
Still an issue here on Arch Linux as well. Forsaken's workaround, with an offset of +1920+100 , worked for Cinnamon.

**Comment by heillon** [comment]
I can't test it any more. Changed monitors and now it opens on the correct screen. (my previous primary now is secondary and the previous secondary is disconnected).

**Comment by joefromct** [comment]
Here's what I did, running a recent version of gnome on arch.

I found an extension called "Auto Move Windows" in Gnome Tweak Tool, and then setup a rule so that any windews generated by steam jump automatically to desktop # 1.  

This seems to work so far, but geez it still feels a bit flaky.  


![image](https://user-images.githubusercontent.com/3342707/104820470-79b0c300-57fa-11eb-803a-f04f7bf82908.png)

**Comment by joefromct** [comment]
In addition to my other comment this is an extension to play with in gnome that might get you were you need to be.   

"Multi Monitors Add-on" and then choose "extend panel to second display"... I think this supports @ForsakenNGS comment above regarding the resolution, and what the panel may be doing to confuse steam big picture.   

![image](https://user-images.githubusercontent.com/3342707/104820696-440cd980-57fc-11eb-9313-2bc4e9daf3ce.png)

**Comment by LiamDawe** [comment]
Still an issue here and makes it a real nuisance, has there been any progress on this? Don't understand why it doesn't launch on the primary - it's set as that for a reason eh.

**Comment by Eschguy** [comment]
Still an issue using POP 21.04. HOW is this still an issue after YEARS?

**Comment by drogeek** [comment]
Still an issue using Ubuntu. Whenever I launch steam link, big picture opens on the screen that is not broadcasted to the TV, I have to manually move it to the other screen every single time

**Comment by Lazaraaus** [comment]
This is still an issue. 
OS: Ubuntu 20.04
Kernel: 5.13.46
DE: KDE Plasma

**Comment by DreamingIntoTheVoid** [comment]
Still an issue with Fedora 36. Which uses the Wayland compositor and Gnome 42.

Where are you're pulling the display info from? Is there something I can alter to make it recognise the correct display? 

I have my right hand monitor set as my primary display in the gnome display configuration.

![image](https://user-images.githubusercontent.com/68373432/164357400-8f032ffd-2273-485f-872a-f2a4533db1be.png)

And this is the result of running xrandr:

```
XWAYLAND0 connected primary 2560x1440+3840+720 (normal left inverted right x axis y axis) 600mm x 340mm
   2560x1440    164.85*+
   1920x1440    164.87  
   ... 
   all available resolutions
   ...
   640x350      164.46  
XWAYLAND1 connected 3840x2160+0+0 (normal left inverted right x axis y axis) 710mm x 400mm
   3840x2160     59.98*+
   ... 
   more resolutions
   ...
   640x350       59.77  
```

**Comment by rockrevenchy** [comment]
I had this happen on a ubuntu + kde plasma.

My own fix was to rightclick the steam big picture window from my taskbar and select "move" then move it to my main monitor.

seems to stay on my main monitor even after restarts.

**Comment by samsunix** [comment]
2024 Ubuntu 23.10 & PopOS 22.04 . This is still the issue after 9 years!!!

**Comment by gabeklavans** [comment]
If using KDE, you can use a Window Rule to force the window matching the title "Steam Big Picture Mode" to open on a specific screen. The exact window title may change, so just open it first and then use the "Detect Window Properties" button in the KDE Window Rules settings to get the proper window title.

**Comment by ChrisLauinger77** [comment]
And in gnome https://extensions.gnome.org/extension/8149/smart-auto-move-ng/

**Comment by bl-ey** [comment]
> If using KDE, you can use a Window Rule to force the window matching the title "Steam Big Picture Mode" to open on a specific screen. The exact window title may change, so just open it first and then use the "Detect Window Properties" button in the KDE Window Rules settings to get the proper window title.

Can you provide your settings? I'm not getting any luck with this. I was able to get it to move once while it was already open, but not I can't get it to effect the Steam BP at all during or after open, it's very strange.

**Comment by BenGrn** [comment]
11 years and still can't open on the primary monitor?
